plugins/crypto/ua_securitypolicy_filestore.c

```diff
@@ -0,0 +1,411 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ *    Copyright 2022 (c) Mark Giraud, Fraunhofer IOSB
+ *    Copyright 2023 (c) Fraunhofer IOSB (Author: Kai Huebl)
+ *    Copyright 2024 (c) Fraunhofer IOSB (Author: Noel Graf)
+ */
+
+#include <open62541/util.h>
+#include <open62541/plugin/securitypolicy_default.h>
+#include <open62541/plugin/log_stdout.h>
+
+#include "mp_printf.h"
+
+#ifdef UA_ENABLE_ENCRYPTION
+
+#ifdef __linux__ /* Linux only so far */
+
+#include <stdio.h>
+#include <dirent.h>
+#include <bits/stdio_lim.h>
+
+typedef struct FileCertStore {
+    /* Memory cert store as a base */
+    UA_SecurityPolicy policy;
```

Comment:
Would it be sufficient, to just store a pointer to the surrounding security policy here?